gatsby-node.js

```diff
@@ -1,14 +1,39 @@
 const path = require("path")
+const { readFileSync } = require("fs")
+const sortLibs = require("./scripts/sort-libraries")
 
 exports.onCreatePage = async ({ page, actions }) => {
   const { createPage, deletePage } = actions
   deletePage(page)
+  let context = {
+    ...page.context,
+    sourcePath: path.relative(__dirname, page.componentPath),
+  }
+  if (page.path === "/code" || page.path === "/code/") {
+    const codeData = JSON.parse(readFileSync("./data/code.json", "utf8"));
+    await Promise.all([
+      Promise.all(Object.keys(codeData.Libraries).map(async languageName => {
+        const libraryCategoryMap = codeData.Libraries[languageName];
+        await Promise.all(
+          Object.keys(libraryCategoryMap).map(async libraryCategoryName => {
+            const libraries = libraryCategoryMap[libraryCategoryName]
+            libraryCategoryMap[libraryCategoryName] = await sortLibs(libraries)
+          })
+        )
+      })),
+      sortLibs(codeData.Tools).then(sortedTools => {
+        codeData.Tools = sortedTools;
+      }),
+    ])
+
+    context = {
+      ...context,
+      codeData,
+    }
+  }
   createPage({
     ...page,
-    context: {
-      ...page.context,
-      sourcePath: path.relative(__dirname, page.componentPath),
-    },
+    context,
   })
```

Comment:
This section again is continuing to get chunks of hard to grok code without the reasoning for why. I think this should probably have comments - and more likely I think that async could probably just be moved to a bootstrapping phrase.